### ISSUE TITLE
Vue 2: ScaleAppShell slots

### ISSUE THREAD

**Issue by SirTomme** [telekom/scale#376]
Hi, I tried using slots in `<scale-app-shell>` in Vue 2 but `header` does not seem to work.

The syntax shown in https://www.brand-design.telekom.com/scale/?path=/docs/components-brand-header-navigation--custom-logo is not specific to Vue because it sets slot content this way: `<template v-slot:header>...</header>`. I am even not sure if Vue supports nested slot templates.

This is what I tried:

```js
<scale-app-shell
  claim-lang="de"
  logo-href="#/home"
  :main-navigation="mainNavigation"
>
  <template v-slot:header>
      <h1>This won't show up anywhere</h1>
  </template>
  <template v-slot:default>
    Anything I change here is displayed on the page.
  </template>
  <!-- <the-notifier /> -->
</scale-app-shell>
```

Any hints how to implement a custom logo or a custom icon menu in Vue 2?

**Comment by acstll** [comment]
Thanks for asking @SirTomme 

The syntax needed is the standard [`slot` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/slot). So nothing specific to Vue.

The value for the `slot` attribute must be the name of the slot you want to replace within the Shadow DOM of the web component. This is what the syntax looks like:

```html
<div slot="header"></div>
```

Here's an example for adding a custom logo and and custom icon menu, in any framework since this is pure HTML:

```html
<scale-app-shell>
  <scale-app-header slot="header">
    <div slot="logo">
      <span>Hello</span>
    </div>
    <div slot="menu-icon">
      <a href="#">A link</a>
    </div>
  </scale-app-header>
</scale-app-shell>
```

You can see we use the `<scale-app-header>` element to fill the `header` slot within `<scale-app-shell>`. And then use a `div` to fill both `logo` and `menu-icon` slots respectively.

Within `scale-app-header` there are the following slots available:

- logo
- menu-main
- menu-icon
- menu-sector
- menu-addon
- menu-mobile

(I can understand the docs can cause some confusion since some Vue syntax is there. We're working on fixing this.)

I hope this helps.

---

If you would like to dig deeper into slot composition, I can highly recommend this article: https://javascript.info/slots-composition

**Comment by zphilipp** [comment]
@SirTomme did you get this to work?

I'm trying to build a custom header in Vue 2 and I can't fill any of the named slots in ScaleAppShell.

Here is my template:
```
<template>
  <scale-app-shell>
    <scale-app-header slot="header">
      <div slot="logo">
        <span>Hello</span>
      </div>
      <div slot="menu-icon">
        <a href="#">A link</a>
      </div>
    </scale-app-header>
    <router-view />
  </scale-app-shell>
</template>
```

I used the example provided by @acstll here and added a router-view.
The router-view is rendered in the default unnamed slot, but the custom header does not work.

Any tips on how to make it work would be great!

**Comment by SirTomme** [comment]
Unfortunately, no. I tried the example but it did not work and I hat not much time to investigate any further.

**Comment by acstll** [comment]
Reopening this to investigate further. Thanks for the feedback @SirTomme 🙏

**Comment by acstll** [comment]
@zphilipp I created a working example in codesandbox with the exact same code snippet:

https://codesandbox.io/s/suspicious-hofstadter-vue-shell-router-w01j2?file=/src/App.vue

Could you maybe provide some more details so we could help you better? For instance:

- what tooling (e.g. vue-cli) are you using
- what version of Vue
- what version of `@telekom/scale-components`
- what errors or screen state exactly are you getting
- any other thing worth sharing

Thank you!

**Comment by zphilipp** [comment]
@acstll After looking over your example I realized that it only works if I use the scale components in my template without importing them inside of the script tag. I forked your example in codesandbox to demonstrate this:

https://codesandbox.io/s/suspicious-hofstadter-vue-shell-router-forked-e0tj0?file=/src/App.vue


Am I not supposed to import and register the components?

Thanks a lot for your help so far!

**Comment by acstll** [comment]
You're welcome, thanks for the feedback!

This looks very much like a bug. The Vue proxy package (or wrapper) is preventing the slots from working properly somehow.

> Am I not supposed to import and register the components?

according to our current docs you could say so, but unlink React (where using the components without the proxy package is almost impossible), Vue perfectly supports web components, so the wrapper is kind of a nice-to-have. Historically, this proxy package (`@telekom/scale-components-vue`) has been more trouble than it has actually helped, and there is no way to support Vue 3 with the current setup. So we decided we're going to deprecate this package (Angular too). We just haven't found the time to actually (and properly) do it.

We're gonna do some more testing, update the boilerplate/examples and update the documentation; hopefully very soon.

So in summary, you don't need to import/register the components, only the `defineCustomElements()` call and `Vue.config.ignoredElements = [/scale-\w*/]` setting are needed.

There are two important things to keep in mind:
- `v-model` is not gonna work, you have to do the bindings manually (here's an example: https://github.com/telekom/scale/issues/402#issuecomment-861250087)
- for attributes that take "data", like an array (header, footer and data-grid), you might need to `JSON.stringify` it

---

Going back to the bug. Since we have plans to deprecate the proxy package, maybe there's no point in hunting for it. Would there be anything preventing you from removing the proxy package from your project?

**Comment by zphilipp** [comment]
That's not a problem. I will just remove the proxies.
Everything seems to work fine, I'll just have to manually do some bindings, as you said.
Thanks again! 😊

**Comment by acstll** [comment]
You're very welcome!
